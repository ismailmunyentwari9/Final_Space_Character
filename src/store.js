import { configureStore } from '@reduxjs/toolkit';
import charactersReducer from './Redux/finalspace/finalspaceSlice';

const store = configureStore({
  reducer: {
    characters: charactersReducer,
  },
});

export default store;
