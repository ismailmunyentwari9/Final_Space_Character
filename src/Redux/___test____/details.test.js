import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom'; // add this import
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import CharacterDetails from '../../components/characterdetails';
import store from '../../store';

it('matches snapshot', () => {
  const mockCharacter = {
    name: 'John Doe',
    gender: 'Male',
    species: 'Human',
  };

  const tree = renderer.create(
    <Router>
      {/* wrap Provider with Router */}
      <Provider store={store}>
        <CharacterDetails selectedCharacter={mockCharacter} />
      </Provider>
    </Router>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
