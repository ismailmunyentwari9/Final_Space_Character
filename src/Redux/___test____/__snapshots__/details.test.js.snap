// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="characterdetails"
>
  <h1>
    <a
      href="/"
      onClick={[Function]}
    >
      <span
        className="to-home"
      >
        <i
          className="fa fa-chevron-left"
        />
      </span>
    </a>
    <img
      alt="character"
    />
    John Doe
  </h1>
  <ul>
    <li>
      <span>
           Gender:
      </span>
      <p>
        Male
        <span>
          <i
            className="fa fa-arrow-right"
          />
        </span>
      </p>
    </li>
    <li>
      <span>
         Status:
      </span>
      <p>
        <span>
          <i
            className="fa fa-arrow-right"
          />
        </span>
      </p>
    </li>
    <li>
      <span>
        Species:
      </span>
      <p>
        Human
        <span>
          <i
            className="fa fa-arrow-right"
          />
        </span>
      </p>
    </li>
    <li>
      <span>
        Hair:
      </span>
      <p>
        <span>
          <i
            className="fa fa-arrow-right"
          />
        </span>
      </p>
    </li>
  </ul>
</div>
`;
